ecl/hthor/hthorkey.cpp

```diff
@@ -1455,16 +1489,33 @@ const void *CHThorIndexCountActivity::nextRow()
                     agent.reportProgress(NULL);
                     if (!klManager->lookup(true))
                         break;
+                    ++keyedProcessed;
+                    if ((keyedLimit != (unsigned __int64) -1) && keyedProcessed > keyedLimit)
```

Comment:
technically the first test is redundant(!), but you can argue that it is clearer with it.